lxc/image.go

```diff
@@ -109,17 +109,29 @@ hash or alias name (if one is set).`))
 	return cmd
 }
 
-func (c *cmdImage) dereferenceAlias(d lxd.ImageServer, imageType string, inName string) string {
+func (c *cmdImage) dereferenceAlias(d lxd.ImageServer, imageType string, inName string) (*api.Image, string, error) {
 	if inName == "" {
 		inName = "default"
 	}
 
-	result, _, _ := d.GetImageAliasType(imageType, inName)
-	if result == nil {
-		return inName
+	result, _, err := d.GetImageAliasType(imageType, inName)
+	if err != nil {
+		// Maybe that inName is a fingerprint and can't be found as an alias
+		image, etag, errImage := d.GetImage(inName)
+		if errImage != nil {
+			return nil, "", fmt.Errorf(i18n.G("Failed fetching fingerprint %q: %w"), inName, errImage)
+		}
+
+		return image, etag, nil
+	}
+
+	// Alias could be resolved, return its image
+	image, etag, err := d.GetImage(result.Target)
+	if err != nil {
+		return nil, "", err
```

Comment:
I suppose we should wrap this in an error message that says `Failed fetching fingerprint %q for alias %q: %w` so we can differentiate it.

Reply:
fixed